From 9e5066a8d808f2e5e9881101b59c24dd9e9c8a2f Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Sun, 5 Apr 2015 10:53:16 +0200
Subject: Warn if wxUSE_STC is set to 1 and in the same time wxUSE_STOPWATCH is
 set to 0.

wxStyledTextCtrl requires wxStopWatch and if it is missing then appropriate actions (warning or correction) are executed in chkconf.h.

diff --git a/include/wx/chkconf.h b/include/wx/chkconf.h
index 92a58b0..963d156 100644
--- a/include/wx/chkconf.h
+++ b/include/wx/chkconf.h
@@ -2258,6 +2258,17 @@
 #   endif
 #endif /* wxUSE_MEDIACTRL */
 
+#if wxUSE_STC
+#   if !wxUSE_STOPWATCH
+#       ifdef wxABORT_ON_CONFIG_ERROR
+#           error "wxStyledTextCtrl requires wxUSE_STOPWATCH"
+#       else
+#           undef wxUSE_STC
+#           define wxUSE_STC 0
+#       endif
+#   endif
+#endif /* wxUSE_STC */
+
 #endif /* wxUSE_GUI */
 
 #endif /* _WX_CHKCONF_H_ */
-- 
1.9.5.msysgit.0

